textdescriptives/components/descriptive_stats.py

```diff
@@ -0,0 +1,80 @@
+"""Calculation of descriptive statistics"""
+from spacy.tokens import Doc
+from spacy.language import Language
+
+import numpy as np
+
+
+@Language.factory("descriptive_stats")
+def create_descriptive_stats_component(nlp: Language, name: str):
+    return DescriptiveStatistics(nlp)
+
+
+class DescriptiveStatistics:
+    def __init__(self, nlp: Language):
+        """Initialise components"""
+        if not Doc.has_extension("token_length"):
+            Doc.set_extension("token_length", getter=self.token_length)
+
+        if not Doc.has_extension("sentence_length"):
+            Doc.set_extension("sentence_length", getter=self.sentence_length)
+
+        if not Doc.has_extension("syllables"):
+            Doc.set_extension("syllables", getter=self.syllables)
+
+        if not Doc.has_extension("counts"):
+            Doc.set_extension("counts", getter=self.counts)
```

Comment:
Again we are kinda missing these for Spans. Where the doc is simply the special case of the entire Span.

Reply:
added in 1cf1527